### DISCUSSION TITLE
PDF | can we import custom font using styles?

### DISCUSSION THREAD

**Discussion by garryatbrighton** [YetiForceCompany/YetiForceCRM#16605] [How to] [answered]
PDF now providing additional dialog "Custom styles". Is it possible to import (from local resources, or for instance google) font and set class which will be respected in PDF output?

**Comment by garryatbrighton** [comment]
Just to add... method introduced in YetiforcePDF package is not working:

`<div data-font data-family="Your Font Name" data-weight="400" data-style="normal" data-file="/absolute/path/to/font/file/YourFontName-Regular.ttf"></div>`

Above div is being stripped off

**Comment by TonyM15** [comment]
Managing different fonts in PDF templater should be more friendly, or at least developed should provide clear guideline on how to implement custom fonts, as all solutions passed here in github are not working (at least for me).

**Comment by s2b-git** [comment]
I did following:
1. 
Downloaded font from https://google-webfonts-helper.herokuapp.com/fonts/kanit?subsets=thai and copied it to:
/vendor/yetiforce/yetiforcepdf/lib/Fonts
2.
Added font to vendor/yetiforce/yetiforcepdf/lib/Objects/Font.php to 'protected static $customFontFiles'
3.
Embed in PDF text:
`<div style="font-family:'Kanit'; font-size:40px; font-weight:200">รายละเอียดงาน</div>`

Result it that font is not printed (garbled)
![image](https://user-images.githubusercontent.com/84110093/197084329-f122df2a-b2b8-4055-b4f2-e25bd233aa00.png)


Indeed help will be appreciated

Best

**Reply by s2b-git** [reply]
![image](https://user-images.githubusercontent.com/84110093/197100378-b7623eb6-e1dc-4302-950f-821bad63e7f2.png)
In PDF editor font shows correctly

**Reply by apcloic** [reply]
Yes, not only with custom fonts, same problem with background-color too, it works in PDF editor but not in PDF output.
#16579

**Reply by waran70** [reply]
> Yes, not only with custom fonts, same problem with background-color too, it works in PDF editor but not in PDF output. #16579

Yeti's charging 120 Euro for paid version so we should expect this thread will be death.

Just one thing... Only for one instance of vtiger I am in charge, I generated appr. 17K USD for developers (last 1 year). I am in charge of 15 instances. What I know, I will not drop a coin to yetis for they rude approach to what they do. People like me dont want things for free, We are looking for devs with common sense, and not those who looking at mirror and self pleasuring themselves with that view . As we all (nearly all) know, the approach of yetis are out of mind. If anyone want details I can throw it here... but it will not be nice.
So my advise is: Yetis, do less but do good. And stop some of your guys commenting, as this is a shame. Change business model approach, and stop relaying open source on your paying clients. This is the most ridiculous thing.
Want money? Develop descent extension and sell it (stop thinking of subscription model - sick american cheating crap). And don't mess with politic - this is not your the best major I guess.

**Reply by apcloic** [reply]
@waran70 So instead of asking yeti's dev to change the way they are, why don't you pay your devs (i mean those who work on your vTiger instances) to fix some issues or add the features you need to Yetiforce ?

**Reply by waran70** [reply]
Answer to your question is clearly formed in my above post, plus: I dont use yetiforce for ANY production related purpose.

**Reply by apcloic** [reply]
No it's definately not. 
I still don't understand why you can't just pay devs, then you have the features you need and so you can use Yetiforce in production.

**Reply by mariuszkrzaczkowski** [reply]
> Answer to your question is clearly formed in my above post, plus: I dont use yetiforce for ANY production related purpose.

I don't understand why you waste everyone's time just complaining when you don't want to use yetiforce

**Comment by OkayashiM** [comment]
Also here I cannot resolve on how to get Japanese characters in PDF output with use of custom font. 

おかし Okayashi

**Comment by waran70** [comment]
yeti introduced paid version of PDF which is not working out of the box with asian chars too

**Comment by garryatbrighton** [comment]
Getting hot but still no solution :)

**Comment by Marshal4582** [comment]
Interested in adding Indian Bahrat font to PDF output. Tried provided here solutions, but no success
tips welcome!

**Comment by s2b-git** [comment, marked as answer]
My way:

Font to be uploaded to: /public_html/vendor/yetiforce/yetiforcepdf/lib/Fonts
Font details to be embed in : /vendor/yetiforce/yetiforcepdf/lib/Objects/Font.php
in func: protected static $fontFiles in the format as an example
' 'saraburn-regular' => [
'400' => [
'normal' => 'Sarabun-Regular.ttf',
],
],'

However this will not show still in CKEditor. Use style such as: 

![image](https://user-images.githubusercontent.com/84110093/197381186-5e38aca0-8df2-4b48-a217-7023aab4b85f.png)

**Reply by garryatbrighton** [reply]
Brilliant
